lib/module.js

```diff
@@ -88,9 +88,18 @@ function readPackage(requestPath) {
 function tryPackage(requestPath, exts) {
   var pkg = readPackage(requestPath);
 
-  if (!pkg) return false;
+  if (!(pkg && pkg.main)) return false;
+
+  var resolved = pkg['_resolved'],
```

Comment:
when running npm install ,  does it not download the code from the _resolved  url   ?   so the _resolved url  must be unique, otherwise things will break badly.